src/cose/decoder.rs

```diff
@@ -129,8 +87,8 @@ pub fn decode_signature(bytes: Vec<u8>, payload: &[u8]) -> Result<Vec<CoseSignat
     let signature_algorithm = match signature_algorithm {
         CborType::SignedInteger(val) => {
             match val {
-                -7 => CoseSignatureType::ES256,
-                -37 => CoseSignatureType::PS256,
+                -7 => SignatureAlgorithm::ES256,
+                -37 => SignatureAlgorithm::PS256,
```

Comment:
I was going to say we should add -35 / ES384 here, but I guess we don't actually support decoding it yet, so let's wait until we have test data for it.

Reply:
yeah, we need some more tests for ES384 (and ES512 if we want to support that).